fish-rust/src/trace.rs

```diff
@@ -0,0 +1,75 @@
+use crate::{
+    common::{escape_string, EscapeFlags},
+    ffi::{self, parser_t, wcharz_t},
+    wchar::{self, wstr, L},
+    wchar_ffi::WCharToFFI,
+};
+use std::sync::atomic;
+
+#[cxx::bridge]
+mod ffi2 {
+    extern "C++" {
+        include!("wutil.h");
+        include!("parser.h");
+        type wcharz_t = super::wcharz_t;
+        type parser_t = super::parser_t;
+    }
+
+    extern "Rust" {
+        fn rust_trace_set_enabled(do_enable: bool);
+        fn rust_trace_enabled(parser: &parser_t) -> bool;
+        fn rust_trace_argv(parser: &parser_t, command: wcharz_t, args: &Vec<wcharz_t>);
+        fn rust_trace_if_enabled(parser: &parser_t, command: wcharz_t);
+    }
+}
+
+static do_trace: atomic::AtomicBool = atomic::AtomicBool::new(false);
```

Comment:
Please use SCREAMING_SNAKE_CASE for static values per rust naming conventions.

Reply:
Sure, wondering why `clippy` or `rustc` don't complain...